matrix/Euler.hpp

```diff
@@ -48,23 +48,13 @@ class Euler : public Vector<Type, 3>
 
     Euler(const Dcm<Type> & dcm) : Vector<Type, 3>()
     {
-        Type psi_val = Type(atan(dcm(1, 0)/ dcm(0, 0)));
-        Type theta_val = Type(asin(-dcm(2,0)));
-        Type phi_val = Type(atan(dcm(2, 1)/ dcm(2, 2)));
-
-        // protection against NaN if dcm(0,0) or dcm(2,2) == 0
-        psi() = 0;
-        theta() = 0;
-        phi() = 0;
-        if (psi() >= -(Type)M_PI_2 && psi() <= (Type)M_PI_2) {
-            psi() = psi_val;
-        }
-        if (theta() >= -(Type)M_PI_2 && theta() <= (Type)M_PI_2) {
-            theta() = theta_val;
-        }
-        if (phi() >= -(Type)M_PI_2 && phi() <= (Type)M_PI_2) {
-            phi() = phi_val;
-        }
+        Type psi_val = Type(atan2f(dcm(1, 0),dcm(0, 0)));
+        Type theta_val = Type(asinf(fminf(fmaxf(-dcm(2,0),-1.0f),1.0f)));
```

Comment:
so this looks like you are bounding the dcm value between -1 and 1, is this typically a problem?